schemas/latest/dbt_yml_files-latest.json

```diff
@@ -0,0 +1,1659 @@
+                "tags": {
+                  "$ref": "#/$defs/string_or_array_of_strings"
+                },
+                "tests": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/$defs/tests"
+                  }
+                }
+              },
+              "additionalProperties": false
+            }
+          },
+          "tags": {
+            "$ref": "#/$defs/string_or_array_of_strings"
+          },
+          "tests": {
+            "type": "array",
+            "items": {
+              "$ref": "#/$defs/tests"
+            }
+          }
+        }
+      }
+    },
+    "unit_tests": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": [
+          "name",
+          "model",
+          "expect"
+        ],
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          },
+          "config": {
+            "type": "object",
+            "properties": {
+              "meta": {
+                "type": "object"
+              },
+              "tags": {
+                "$ref": "#/$defs/string_or_array_of_strings"
+              }
+            },
+            "additionalProperties": false
+          },
+          "expect": {
+            "type": "object",
+            "if": {
+              "properties": {
+                "format": {
+                  "const": "dict"
+                }
+              }
+            },
+            "then": {
+              "properties": {
+                "rows": {
+                  "description": "When `format` is `dict`, each item should be a dictionary containing a key-value pair for each column and its value, e.g. `{id: 1, code: 'ABC'}`",
+                  "type": "array",
+                  "items": {
+                    "type": "object"
+                  }
+                }
+              }
+            },
+            "else": {
+              "properties": {
+                "rows": {
+                  "description": "When `format` is csv, a string containing comma-separated headers and values. Alternatively provide a fixture. Use the pipe character | to create a multi-line string in YAML.",
+                  "type": "string"
+                }
+              }
+            },
+            "properties": {
+              "fixture": {
+                "description": "Only relevant when format is csv. Specify the name of a fixture instead of providing `rows`.",
+                "type": "string"
+              },
+              "format": {
+                "description": "Defaults to `dict` when not specified",
+                "type": "string",
+                "enum": [
+                  "dict",
+                  "csv"
+                ]
+              },
+              "additionalProperties": false
+            }
+          },
+          "given": {
+            "type": "array",
```

Comment:
is it worth moving the logic around data to a `def` and reusing for `given` and `expect`?

Reply:
It surely is

Reply:
Actually I don't think it's possible - `given` includes an `input` block, and is also an array of objects instead of a single object. 

As far as I can see, I can't union all of the common stuff with the `input` block for the `given` schema.